### PR TITLE
rework `@test_reference` messages - drop lena

### PR DESCRIPTION
**Pull request by t-bltg**
- rework `@info` in `@test_reference`: now includes `render(...)` (more structured output for the end user, see below, works well with images), this helps CI in ensuring correct message order (io flush);
- ~~rework tests not to show spurious messages when working with `@test_reference` macro unless `CI` env var is set to `true`: this was is very confusing when running `] test ReferenceTests`: only the tests summary should be shown to the end user, detailed are activated in CI or by the env var `CI`;~~
- [drop](https://en.wikipedia.org/wiki/Lenna#Criticism) `lena` in `runtests.jl`: replaced with `monarch` which is very similar (same format and size);
- stop writing / deleting temporary test files in `test/references`: if testing is interrupted, these temporaries are leftovers, and thus can land in a commit (use `mktempdir` instead).

As an example, [this job output](https://github.com/JuliaTesting/ReferenceTests.jl/runs/7787991354?check_suite_focus=true#step:5:525) (PR) is more readable than [this one](https://github.com/JuliaTesting/ReferenceTests.jl/runs/7787894924?check_suite_focus=true#step:5:1626) (master): most likely flush of `stdout` and `stderr` is de-synchronized.

**PR**
```julia
julia> ENV["VERBOSE"] = true  # default

pkg> test ReferenceTests
[...]
Test Summary:    | Pass  Total
world age issues |    1      1
┌ Info: Reference Test for "string1.txt" failed:
│ - DIFF ------------------------
│ "fintentionally wrobng to check that this message prints"
│ -------------------------------
│   reference = "[...]/test/references/string1.txt"
└   actual = "/tmp/jl_h48D23/string1.txt"
┌ Info: Reference Test for "string5.txt" failed:
│ - DIFF ------------------------
│ """
│ - This is a
│ + This is an incorrect
│   multiline string that does not end with a new line."""
│ -------------------------------
│   reference = "[...]/test/references/string5.txt"
└   actual = "/tmp/jl_79MGXW/string5.txt"
[...]
     Testing ReferenceTests tests passed

julia> ENV["VERBOSE"] = false

pkg> test ReferenceTests
[...]
Test Summary:    | Pass  Total
world age issues |    1      1
Test Summary:  | Pass  Total
ReferenceTests |   49     49
     Testing ReferenceTests tests passed 
```

**master**
```julia
pkg> test ReferenceTests
[...]
┌ Info: Reference Test for "string1.txt" failed.
│   reference = "[...]/test/references/string1.txt"
└   actual = "/tmp/jl_egX948/string1.txt"
- DIFF ------------------------
"fintentionally wrobng to check that this message prints"
-------------------------------
┌ Info: Reference Test for "string5.txt" failed.
│   reference = "[...]/test/references/string5.txt"
└   actual = "/tmp/jl_qicFon/string5.txt"
- DIFF ------------------------
"""
- This is a
+ This is an incorrect
  multiline string that does not end with a new line."""
-------------------------------
[...]
     Testing ReferenceTests tests passed 
```

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -3,16 +3,26 @@ using FileIO
 using BSON: BSON
 using Random
 using ReferenceTests
+using Logging
+
+ci = tryparse(Bool, get(ENV, "CI", "false")) === true && !isinteractive()
+@show ci
+
+# minimize messages when testing locally, all messages on CI
+test_logger = ci ? current_logger() : NullLogger()
```
Comment (oxinabox) [inline test/runtests.jl:12]: I am still not convinced of the wisdom of this.
What do you think @johnnychen94 ?

Comment (johnnychen94) [reply test/runtests.jl:12]: Agreed, I feel it's okay to let the test of ReferenceTest be verbose by default and it won't hurt for users of it. After all, this is strictly an unrelated change and is better to be avoided.

**Review by johnnychen94 — APPROVED** [review]

File: test/runtests.jl (hunk 1 above)
Comment (t-bltg) [reply test/runtests.jl]: @oxinabox, I have restored the verbose option by default.

**Comment by t-bltg** [comment]
@oxinabox, I have some issues with nightly CI in UnicodePlots (see e.g. https://github.com/JuliaPlots/UnicodePlots.jl/actions/runs/7300585284/job/19895531268#step:6:384).

This PR makes the output from `ReferenceTests` clearer for debugging.

Can we please revive and merge this PR and bump a new release ?

**Comment by t-bltg** [comment]
CI is green for all OS.

**Review by oxinabox — APPROVED** [review]

**Pull request merged**